RecoTracker/MkFit/plugins/createPhase1TrackerGeometry.cc

```diff
@@ -0,0 +1,372 @@
+      sp.append_plan(18, false);
+      sp.append_plan(19, false);
+      sp.append_plan(20, false);
+      sp.fill_plan(21, 26);  // TID,  6 layers
+      sp.fill_plan(27, 44);  // TEC, 18 layers
+      sp.finalize_plan();
+    }
+  }
+
+  void setupIterationParams(IterationParams &ip, unsigned int it = 0) {
+    if (it == 0) {
+      ip.nlayers_per_seed = 4;
+      ip.maxCandsPerSeed = 5;
+      ip.maxHolesPerCand = 4;
+      ip.maxConsecHoles = 1;
+      ip.chi2Cut = 30;
+      ip.chi2CutOverlap = 3.5;
+      ip.pTCutOverlap = 1;
+    } else if (it == 1)  // for triplet steps, nlayers_per_seed=3
+    {
+      ip.nlayers_per_seed = 3;
+      ip.maxCandsPerSeed = 5;
+      ip.maxHolesPerCand = 4;
+      ip.maxConsecHoles = 1;
+      ip.chi2Cut = 30;
+      ip.chi2CutOverlap = 3.5;
+      ip.pTCutOverlap = 1;
+    } else if (it == 2) {
+      ip.nlayers_per_seed = 4;
+      ip.maxCandsPerSeed = 5;
+      ip.maxHolesPerCand = 4;
+      ip.maxConsecHoles = 1;
+      ip.chi2Cut = 30;
+      ip.chi2CutOverlap = 3.5;
+      ip.pTCutOverlap = 1;
+    } else if (it == 3)  // for triplet steps, nlayers_per_seed=3
+    {
+      ip.nlayers_per_seed = 3;
+      ip.maxCandsPerSeed = 5;
+      ip.maxHolesPerCand = 4;
+      ip.maxConsecHoles = 1;
+      ip.chi2Cut = 30;
+      ip.chi2CutOverlap = 3.5;
+      ip.pTCutOverlap = 1;
+    } else if (it == 4) {
+      ip.nlayers_per_seed = 4;
+      ip.maxCandsPerSeed = 5;
+      ip.maxHolesPerCand = 4;
+      ip.maxConsecHoles = 1;
+      ip.chi2Cut = 30;
+      ip.chi2CutOverlap = 3.5;
+      ip.pTCutOverlap = 1;
+    } else if (it == 5)  // for triplet steps, nlayers_per_seed=3
+    {
+      ip.nlayers_per_seed = 3;
+      ip.maxCandsPerSeed = 5;
+      ip.maxHolesPerCand = 4;
+      ip.maxConsecHoles = 1;
+      ip.chi2Cut = 30;
+      ip.chi2CutOverlap = 3.5;
+      ip.pTCutOverlap = 1;
+    } else if (it == 6)  // for triplet steps, nlayers_per_seed=3; for mixeTripletSetp, also maxCandsPerSeed=2
+    {
+      ip.nlayers_per_seed = 3;
+      ip.maxCandsPerSeed = 2;
+      ip.maxHolesPerCand = 4;
+      ip.maxConsecHoles = 1;
+      ip.chi2Cut = 30;
+      ip.chi2CutOverlap = 3.5;
+      ip.pTCutOverlap = 1;
+    } else if (it == 7)  // for PixelLess step, maxCandsPerSeed=2 and maxHolesPerCand=maxConsecHoles=0
+    {
+      ip.nlayers_per_seed = 3;
+      ip.maxCandsPerSeed = 2;
+      ip.maxHolesPerCand = 0;
+      ip.maxConsecHoles = 0;
+      ip.chi2Cut = 30;
+      ip.chi2CutOverlap = 3.5;
+      ip.pTCutOverlap = 1;
+    } else if (it == 8)  // for TobTec step, maxCandsPerSeed=2 and maxHolesPerCand=maxConsecHoles=0
+    {
+      ip.nlayers_per_seed = 3;
+      ip.maxCandsPerSeed = 2;
+      ip.maxHolesPerCand = 0;
+      ip.maxConsecHoles = 0;
+      ip.chi2Cut = 30;
+      ip.chi2CutOverlap = 3.5;
+      ip.pTCutOverlap = 1;
+    }
+  }
+
+  void fillHitSelectionWindowsParams(IterationConfig &ic) {
+    HitSelectionWindowsPhase1 hsw;
+    for (int l = 0; l < (int)ic.m_layer_configs.size(); ++l) {
```

Comment:
`auto` ?

Reply:
To me `auto` doesn't make practical difference. On the other hand, something along
```suggestion
    for (int l = 0, end = ic.m_layer_configs.size(); l < end; ++l) {
```
would make.

Reply:
This function was removed.